stack.py

```diff
@@ -88,16 +93,19 @@ def eval_program(program):
                 a = stack.pop()
             # b is the top of the stack, and a is the item before it, so
             # `... ; push 5 ; div` is dividing the result of `...` by 5.
-            if instr.op == 'add':
-                c = a + b
-            elif instr.op == 'sub':
-                c = a - b
-            elif instr.op == 'mul':
-                c = a * b
-            elif instr.op == 'div':
-                c = a / b
-            elif instr.op == 'pow':
-                c = a ** b
+            binary_ops = {
```

Comment:
NICE, this is exactly the refactoring I was going to suggest.
One final thing, I don't think it belongs inside the function, move it out into the module's scope.